string path_name
uint16 number
string[] nav_name
string[] task_name